### PR TITLE
Refactor start_proteus.py and move config to toml

### LINKED ISSUE #141: Refactor `start_proteus.py`
As discussed at yesterdays meeting, I'm opening this issue to track the progress with refactoring `start_proteus.py`.

The goal is to move all the logic etc into the library so that it can be re-used elsewhere. The goal is to have 3 entry points:

1. `start_proteus.py`, this will work in the same way as before. It will just be a lot smaller. This can be used to specify config variables, paths etc, but the main code will be elsewhere.
2. `proteus start`, this will be a CLI tool that starts proteus. The advantage is that it can be started from anywhere and does not require the user to git clone the repo. This will be more relevant as we move away from the submodules.
3. Python api, e.g. in a jupyter notebook. This is somewhat similar to 1. This will enable more exploratory use of the code, e.g. by importing the proteus class into your code: 

```python
from proteus import Proteus

handler = Proteus(cfg='input/default.cfg')
handler.start()
```

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR refactors the `start_proteus.py`. This includes an update to the CLI, in line with what we did for janus / mors.

```console
proteus start --config input/default.toml
```

or

```console
python start_proteus.py
```

I also started to move the config to toml. Moving to toml was fairly straightforward and only meant minor changes in the cfg files itself. Many parts of the code read the config directly, which is turned out to be more work than I had anticipated.

Next steps would be to break down the `Proteus.start()` function into isolated methods, but I want to tackle that in a follow-up PR to avoid making too many changes at once.

## TODO
- [x] Update `plot/cpl_*` files to use toml
- [x] Update `utils/plot_offchem.py` to use toml
- [x] Add tests
- [x] Fix CI workflow

Closes #141 
Addresses #74

### CODE REVIEW

**Comment by stefsmeets** [comment]
Hi @lsoucasse @nichollsh Could you please review this PR and let me know what you think?

The main files to look at would be `src/proteus/proteus.py`, `src/proteus/cli.py`, `src/proteus/utils/coupler.py`.

Because of the toml change, I also had to update all the code that reads in the config directly, see `src/proteus/plot`, and specifically `_cpl_helpers.py` for how I refactored it.

Happy to talk you through the PR if necessary.

**Comment by nichollsh** [comment]
Looks good. These changes make sense to me, and it's nice to have a consistent CLI for PROTEUS.

However, doesn't this make `start_proteus.py` redundant? Why include it at all if there's already the `click` interface for running the model via `proteus start`?

File: docs/usage.md [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
 ```console
-python start_proteus.py --cfg [cfgfile]
+proteus start --config [cfgfile]
 ```
 
 Where `[cfgfile]` is the path to the required configuration file. If
-`--cfg [cfgfile]` is not provided, then the default configuration
-located at `input/default.cfg` will be used. Pass the flag `--resume` in
+`--config [cfgfile]` is not provided, then the default configuration
+located at `input/default.toml` will be used. Pass the flag `--resume` in
```
Comment (lsoucasse) [inline docs/usage.md:13]: The config file is always expected by the cli function.

**Review by lsoucasse — CHANGES_REQUESTED** [review]
Thanks @stefsmeets, Nice to switch to toml and to have the cli for running proteus. I agree for now the start_proteus.py script is redundant but once we split the big running function into different methods it might be worth having it.
I tested all the commands, it seems to run fine. Just need to update the documentation because the path to config file is mandatory for the cli.

**Comment by stefsmeets** [comment]
Cheers both, agree on the `start_proteus.py`. I kept it, because it signals how to call the code from Python. And it is part of the current workflow for some people. I think it does not hurt to keep it around for a while, but if you both think we can remove it we can also remove it.

**Comment by lsoucasse** [comment]
> Cheers both, agree on the `start_proteus.py`. I kept it, because it signals how to call the code from Python. And it is part of the current workflow for some people. I think it does not hurt to keep it around for a while, but if you both think we can remove it we can also remove it.

I think we can keep it for now.

**Comment by stefsmeets** [comment]
@nichollsh @lsoucasse Could you check again and approve? This PR is ready to be merged.

**Review by lsoucasse — APPROVED** [review]
Good for me.

**Pull request merged**